### PR TITLE
add mdn attachments cdn

### PR DESCRIPTION
**Pull request by metadave**
See https://github.com/mozmeao/infra/issues/590

### CODE REVIEW

**Comment by metadave** [comment]
Please verify these values:

- `response_page_path`
- `default_ttl`
- `max_ttl`

File: apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  comment         = "${var.comment}"
+  enabled         = true
+  http_version    = "http2"
+  is_ipv6_enabled = false
+  price_class     = "PriceClass_All"
+
+  custom_error_response {
+    error_caching_min_ttl = 300
+    error_code            = 404
+    response_page_path    = "/404.html"
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf]: This page doesn't exist:

https://mdn.mozillademos.org/404.html

I think the line should be removed.

File: apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf [hunk 2]
```diff
@@ -32,3 +33,15 @@ module "mdn-cloudfront-prod" {
 }
 
 
+# Attachments origin
+
+module "mdn-cloudfront-attachments-prod" {
+    source = "./cloudfront_attachments"
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf]: I guess this is why the terraform file has to be in a subdirectory

File: apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf [hunk 3]
```diff
@@ -32,3 +33,15 @@ module "mdn-cloudfront-prod" {
...
+
+    acm_cert_arn = "arn:aws:acm:us-east-1:236517346949:certificate/4322464c-b6d9-44d9-914d-3554461bcbb3"
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf]: ``us-east-1``? Is this right?

Comment (metadave) [reply apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf]: all cloudfront certs have to be provisioned in us-east-1.

File: apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf (hunk 2 above)
Comment (metadave) [reply apps/mdn/mdn-aws/infra/mdn-cdn/mdn_cdn.tf]: yes, it's just the subdir name for additional terraform files. It helps with Terraform reuse.

File: apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf (hunk 1 above)
Comment (metadave) [reply apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf]: :+1:

**Comment by jwhitlock** [comment]
Oops, forgot to look at ttl values. Reading the docs...

File: apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    error_code            = 404
+  }
+
+  default_cache_behavior {
+    allowed_methods = ["GET", "HEAD", "OPTIONS"]
+    cached_methods  = ["GET", "HEAD", "OPTIONS"]
+    compress        = true
+
+    default_ttl     = 86400
+    max_ttl         = 31536000
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf]: In https://github.com/mozilla/kuma/pull/4476, we're setting the ``Cache-Control`` headers for most responses, so [according to the docs](http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html) we'll get the lesser of ``max_ttl`` and the ``max-age`` we're sending.  For some we're sending 5 minutes (60 * 5 = 300), others 5 days (60 * 60 * 24 * 5 = 432000). I think we're not sending caching headers on some error responses.

I'm more comfortable with 24 hours for default (86400, current setting), and 5 days for max (432000). 1 year seems like asking for trouble for the next devs.

Comment (metadave) [reply apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf]: fixed in the latest push.

btw, the 1 year value came from the SCL3 Cloudfront config :-)

Comment (jwhitlock) [reply apps/mdn/mdn-aws/infra/mdn-cdn/cloudfront_attachments/cloudfront_attachments.tf]: 😳

**Comment by metadave** [comment]
Certs have already been updated on prod ELBs in Portland && Frankfurt

**Comment by metadave** [comment]
terraform has been applied

**Pull request merged**